packages/wdio-config/tests/configparser.test.ts

```diff
@@ -728,39 +728,55 @@ describe('ConfigParser', () => {
             const configParser = await ConfigParserForTest()
             await configParser.addConfigFile(FIXTURES_CONF)
             configParser.merge({ spec: [INDEX_PATH, FIXTURES_CONF] })
-            configParser.merge({ exclude: [path.resolve(__dirname).replace(/\\/g, '/') + '/*'] })
-            const specs = configParser.getSpecs()
-            expect(specs).toHaveLength(2)
+
+            // first validate that the conf fixture 'spec' was merged successfully
+            expect(configParser.getSpecs()).toHaveLength(1)
+
+            configParser.merge({ exclude: [path.join(__dirname, '**', '*conf*')] })
+
+            // then after merging an exclude containing a glob pattern, validate that the exclude
+            // attribute contains multiple items and the filtering on the spec attribute works
+            expect(configParser.getConfig().exclude.length).toBeGreaterThan(0)
+            expect(configParser.getSpecs()).toHaveLength(0)
         })
 
-        it('should overwrite exclude if piped into cli command', async () => {
+        it('should overwrite config exclude if piped into cli command', async () => {
             const configParser = await ConfigParserForTestWithAllFiles()
             await configParser.addConfigFile(FIXTURES_CONF)
+            // check that the initial exclude from the config filtered out 1 spec (out of a total of 4)
+            expect(configParser.getSpecs()).toHaveLength(3)
+
             configParser.merge({ exclude: [INDEX_PATH] })
-            const specs = configParser.getSpecs()
-            expect(specs).toHaveLength(4)
+            // after overriding the config exclude with cli exclude check that the initial config exclude is discarded
+            expect(configParser.getSpecs()).toHaveLength(4)
         })
 
-        it('should overwrite exclude if piped into cli command with params', async () => {
+        it('should overwrite config and capabilities exclude if piped into cli command', async () => {
```

Comment:
it looks to me as if `getFilePaths()` has a hidden bug:
```
ConfigParser.getFilePaths(['wdio-config/src/index.ts', 'wdio-config/src/constants.ts',
    'wdio-config/src/types.ts', 'wdio-config/src/utils.ts'], undefined, this._pathService)
```
only returns the `index.ts` file which is problematic to all tests using this because they can allow bugs to pass uncaught by the unit tests (i tested this on the latest `main` branch).

I only refactored the tests which were related to take the above mentioned into account.

Reply:
I kept having this issue when writing unit tests as well. It made it extremely difficult to figure out what was going wrong.

Reply:
maybe some veteran could help us figure out whats going on here, if not I will just create a ticket to be investigated separately

Reply:
The configparser tests use a mocked file system which makes writing tests around it a bit difficult. Ideally I would love to rewrite this but this is how things are atm, see https://github.com/webdriverio/webdriverio/blob/6a4dd4402e82b6bf87ed802adb0883b08b16dffd/packages/wdio-config/tests/configparser.test.ts#L138-L150